cas-server-support-oauth/src/main/resources/META-INF/spring/cas-servlet-oauth.xml

```diff
@@ -14,12 +14,29 @@
        <context:component-scan base-package="org.pac4j.springframework.web" />
 
        <bean id="oauthSecConfig" class="org.pac4j.core.config.Config"
-             c:callbackUrl="${server.prefix:http://localhost:8080/cas}/oauth2.0/callbackAuthorize"
-             c:client-ref="oauthCasClient" />
+             c:callbackUrl="${server.prefix:http://localhost:8080/cas}/oauth2.0/callbackAuthorize">
+              <constructor-arg name="clients">
+                     <array>
+                            <ref bean="oauthCasClient" />
+                            <ref bean="clientBasicAuth" />
+                            <ref bean="clientForm" />
+                            <ref bean="userForm" />
+                     </array>
+              </constructor-arg>
+       </bean>
 
        <bean id="oauthCasClient" class="org.pac4j.cas.client.CasClient" p:casLoginUrl="${server.prefix:http://localhost:8080/cas}/login"
              p:name="CasOAuthClient" />
 
+       <bean id="clientBasicAuth" class="org.pac4j.http.client.direct.DirectBasicAuthClient" p:authenticator-ref="oAuthClientAuthenticator"
```

Comment:
The resource owner password grant type is ensured by the appropriate security configuration on the /accesstoken endpoint.